Makefile

```diff
@@ -27,6 +28,8 @@ include .make.defaults
 # Global rules that are generally to be implemented in the sub-directories and can
 # be overridden there (the double colon on the rule makes the overridable). 
 
+all: clean test build
```

Comment:
I don't know if we're ready for this, in particular `build` is ill-defined.

Reply:
@daw3rd and @roytman If making a `make all` target is not feasible, please close this PR, but before doing that, Dean has found a typo in the `setup.md` file: **deside => decide**